subsys/net/lib/http/http_server_core.c

```diff
@@ -873,7 +873,21 @@ int http_server_sendall(struct http_client_ctx *client, const void *buf, size_t
 	while (len) {
 		ssize_t out_len = zsock_send(client->fd, buf, len, 0);
 
-		if (out_len < 0) {
+		if ((out_len == 0) || (out_len < 0 && errno == EAGAIN)) {
```

Comment:
Why would we expect `EAGAIN` on a blocking socket with no timeout set?

Reply:
@rlubos 
As for the timeout, I think you are right, we do need a timeout here.